README.md

```diff
@@ -70,13 +70,13 @@ Configuration of type `Application`, the main class is
 To run the server side unit tests in intelliJ, right click on the `AgentDiscoveries-Backend/src/test/java` folder and then choose `Run 'All Tests'`.
 From the command line, use `mvn test`. 
 
-Unit test files should be named ending with "Test.java", whilst end-to-end tests should have a "IT.java" name ending (since the system views them as integration tests).
+Unit test files should be named ending with "Test.java".
 
 ## Frontend Tests
 To run the front-end unit tests, from command line navigate to `AgentDiscoveries-Frontend` and then run npm test.
 
 ## End-to-End Tests
 In order to run the End-To-End tests locally the webserver needs to be active, so ensure you first run the application as described in the "Run" section.
```

Comment:
It would be nice to automate this if possible - particularly so they can run on Jenkins.

Reply:
As above, Jenkins is running the java end-to-end tests automatically, and I've configured this branch to run the node ones on the same command, so hopefully it would work but untested

Reply:
Oh I see what's going on here! - The Jenkins tests actually hit the deployed site.

I think that's really cool, and provides an additional check that the deployment and infrastructure works,
however IMO we should also be able to the run in an integration-test mode first.

Running integration tests before deploying (against a server running on Jenkins itself) gives a few benefits:
- Tighter control of the environment - they won't fail because somebody created some dodgy data on the deployed site.
- Faster feedback - the shorter the feedback loop, the happier developers are to fix bugs!
- Avoid broken deployments - discovering fatal errors _before_ deploying means the site should always be in a (relatively) good state!

You could include both, to give a process like:

- Build
- Unit Test
- Package
- Integration Test
- Deploy
- End-to-End Test